### PR TITLE
Fix xml concatenate behaviour

### PR DESCRIPTION
**Pull request by anupama-pathirage**
Fixes https://github.com/ballerinalang/ballerina/issues/3573

**Current behavior** - When datatable is converted to xml, it does not build the xml data in the memory. So even if that xml is used with + operator the xml is not build and it will be build only when it is consumed later. So the DB connection is held open until it is consumed.

**Fix** - If the xml is accessed or modified via any operator or method, it should build in memory. So it will free up all the DB resources after that.  If someone needs the streamed behavior for the converted xml(without loading into the memory), he needs to send it without doing any modifications or operations with it.

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BXML.java [hunk 1]
```diff
@@ -276,6 +276,9 @@ public String getMessageAsString() {
     public BallerinaMessageDataSource clone() {
         return copy();
     }
+
+    public void build() {
```
Comment (SupunS) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BXML.java]: We can make this abstract.

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/util/XMLUtils.java [hunk 2]
```diff
@@ -159,6 +159,7 @@
                 concatSeq.add(j++, seq.get(i));
```
Comment (SupunS) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/util/XMLUtils.java]: I think each `seq.get(i)` should also get built. We can implement the `build()` in XMLSequence as well, which will recursively visit its children and build them.

Then we can call `firstSeq.build();` and `secondSeq.build();` at the start of this (`concatentae(...)`) method.. WDYT?

Comment (anupama-pathirage) [reply modules/ballerina-core/src/main/java/org/ballerinalang/model/util/XMLUtils.java]: Fixed. Thanks.

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BXML.java (hunk 1 above)
Comment (anupama-pathirage) [reply modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BXML.java]: Fixed. Thanks.

**Pull request merged**